etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiMessageHelper.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    public String extractSendingFacilityNamespace(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(SENDING_FACILITY_NAMESPACE));
+    }
+
+    public String extractSendingFacilityUniversalId(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(SENDING_FACILITY_UNIVERSAL_ID));
+    }
+
+    public String extractSendingFacilityUniversalIdType(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(SENDING_FACILITY_UNIVERSAL_ID_TYPE));
+    }
+
+    public String extractReceivingFacilityNamespace(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(RECEIVING_FACILITY_NAMESPACE));
+    }
+
+    public String extractReceivingFacilityUniversalId(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(RECEIVING_FACILITY_UNIVERSAL_ID));
+    }
+
+    public String extractReceivingFacilityUniversalIdType(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(RECEIVING_FACILITY_UNIVERSAL_ID_TYPE));
+    }
+
+    public String extractSendingApplicationNamespace(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(SENDING_APPLICATION_NAMESPACE));
+    }
+
+    public String extractSendingApplicationUniversalId(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(SENDING_APPLICATION_UNIVERSAL_ID));
+    }
+
+    public String extractSendingApplicationUniversalIdType(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(SENDING_APPLICATION_UNIVERSAL_ID_TYPE));
+    }
+
+    public String extractReceivingApplicationNamespace(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(RECEIVING_APPLICATION_NAMESPACE));
+    }
+
+    public String extractReceivingApplicationUniversalId(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(RECEIVING_APPLICATION_UNIVERSAL_ID));
+    }
+
+    public String extractReceivingApplicationUniversalIdType(Bundle messageBundle) {
+        return fhirEngine.getStringFromFhirPath(
+                messageBundle, fhirPaths.get(RECEIVING_APPLICATION_UNIVERSAL_ID_TYPE));
+    }
+
+    private Map<String, String> loadFhirPaths() {
+        Map<String, String> tempPaths = new HashMap<>();
+        tempPaths.put(
+                RECEIVING_FACILITY_NAMESPACE,
+                """
+            Bundle.entry.resource.ofType(MessageHeader).destination.receiver.resolve().identifier.where(
```

Comment:
I think it'd be good elevate this fhirpaths into static strings at the beginning of the class. I'm thinking we could refactor later so they are accessible in other parts of the project and we have a central place to define them

Reply:
We took the map out and went with this approach. Good suggestion!